test_coverage/test_gcovr.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+# source code: https://github.com/gcovr/gcovr/
+# docs: https://gcovr.com/en/stable/guide.html
+# install GCOVR
+# $ pip install gcovr
+
+PREFIX_ERRN="ERRN:"
+DIR_BUILD="build_gcovr"
+TIMESTAMP=$(date +"%y-%m-%d-%H:%M:%S")
+DIR_REPORT="test_coverage/report_gcovr_${TIMESTAMP}"
+# Jump to build folder
+if [ ! -d ../$DIR_BUILD ]
+  then
+    mkdir ../$DIR_BUILD
+    if [ $? -ne 0 ]
+    then
+      echo $PREFIX_ERRN "cannot make dir '"$DIR_BUILD"'"
+      exit 1
+    fi  
+fi
+cd ../$DIR_BUILD
+if [ $? -ne 0 ]
+  then
+  echo $PREFIX_ERRN "cannot change dir '"$DIR_BUILD"'"
+  exit 1
+fi
+rm -rf ./*
+# Build project with flags
+cmake -DFACELIFT_ENABLE_GCOV=ON .. && make
+# Run facelift test-cases
+examples/launch-tuner.sh
+# Create folder for reports
+mkdir -p ../$DIR_REPORT
+if [ $? -ne 0 ]
+  then
+  echo $PREFIX_ERRN "cannot make dir '"$DIR_REPORT"'"
+  exit 1
+fi
+# Processing GCOVR
+gcovr -r ../ . --html --html-details -o ../$DIR_REPORT/report.html
+cd ../$DIR_REPORT
+if [ $? -ne 0 ]
+  then
+  echo $PREFIX_ERRN "cannot change dir '"$DIR_REPORT"'"
+  exit 1
+fi
+# Show index report
+firefox report.html
```

Comment:
delete this. Print "Test report generated: report.html" (or similar message) instead